core/touch_gesture.js

```diff
@@ -257,28 +257,33 @@ Blockly.TouchGesture.prototype.handleTouchMove = function(e) {
   // Update the cache
   this.cachedPoints_[pointerId] = this.getTouchPoint(e);
 
-  var pointers = Object.keys(this.cachedPoints_);
-  // If two pointers are down, check for pinch gestures
-  if (pointers.length == 2) {
-    // Calculate the distance between the two pointers
-    var point0 = this.cachedPoints_[pointers[0]];
-    var point1 = this.cachedPoints_[pointers[1]];
-    var moveDistance = Blockly.utils.Coordinate.distance(point0, point1);
-    var startDistance = this.startDistance_;
-    var scale = this.touchScale_ = moveDistance / startDistance;
+  var workspace = this.startWorkspace_;
+  // If pinch option is enabled
+  if (workspace.options.zoomOptions && workspace.options.zoomOptions.pinch) {
+    var pointers = Object.keys(this.cachedPoints_);
+    // If two pointers are down, check for pinch gestures
+    if (pointers.length == 2) {
+      // Calculate the distance between the two pointers
+      var point0 = this.cachedPoints_[pointers[0]];
+      var point1 = this.cachedPoints_[pointers[1]];
+      var moveDistance = Blockly.utils.Coordinate.distance(point0, point1);
+      var startDistance = this.startDistance_;
+      var scale = this.touchScale_ = moveDistance / startDistance;
```

Comment:
I know this already existed, but I'm not a fan of chaining assignment like this.  I prefer just setting `this.touchScale_` and using it in place of `scale` in the two locations below.